pydasi/dasi/key.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023 European Centre for Medium-Range Weather Forecasts (ECMWF)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ._dasi_cffi import ffi, ffi_decode, ffi_encode, lib
+from .utils import DEBUG, getLogger
+
+logger = getLogger(__name__)
+logger.setLevel(DEBUG)
+
+
+class Key:
+    """
+    Container for keyword:value pairs.
+    """
+
+    def __init__(self, pair=None):
+        if isinstance(pair, Key):
+            self = pair.copy()
```

Comment:
This code doesn't do what you think it does. (Self as a variable only has scope of the local function - so this creates and then destroys a copy of "pair".

I'm not really sure what you are trying to do with this?

Reply:
done